src/main/java/fi/metatavu/metamind/server/rest/StoriesApiImpl.java

```diff
@@ -160,20 +190,48 @@ public Response createIntent(Intent body, UUID storyId) {
     IntentTrainingMaterial intentRegexTrainingMaterial = trainingMaterialController.setIntentTrainingMaterial(intent, TrainingMaterialType.INTENTREGEX, intentRegexMaterial);
     IntentTrainingMaterial variableOpenNlpRegexTrainingMaterial = trainingMaterialController.setIntentTrainingMaterial(intent, TrainingMaterialType.VARIABLEOPENNLPREGEX, variableOpenNlpRegexMaterial);
     
+    try {
+      UUID createdResourceId = authenticationController.createProtectedResource(loggedUserId, String.format(INTENT_NAME_BEGINNING, intent.getId()), String.format("/v2/stories/intents/%s", intent.getId()), "intent", SCOPES);
+      List<UUID> policyIds = authenticationController.updatePermissionUsers(REALM_NAME);
+      authenticationController.upsertScopePermission(REALM_NAME, createdResourceId, SCOPES, String.format("Permission for intent-%s", intent.getId()), DecisionStrategy.AFFIRMATIVE, policyIds);
+      Set<UUID> permittedUsersId = authenticationController.getResourcePermittedUsers(REALM_NAME, createdResourceId, String.format(INTENT_NAME_BEGINNING, intent.getId()), SCOPES);
+      if (!permittedUsersId.contains(loggedUserId) || loggedUserId == null) {
+        return createBadRequest(String.format(NO_USER_FOUND_MESSAGE));
+      }
+    } catch (Exception e) {
+      logger.warn(String.format(EXCEPTION_CAUGHT, e.getMessage()));
+    }
+    
     return createOk(intentTranslator.translateIntent(intent, intentOpenNlpDocatTrainingMaterial, variableOpenNlpNerTrainingMaterial, intentRegexTrainingMaterial, variableOpenNlpRegexTrainingMaterial));
   }
 
   @Override
   public Response createKnot(Knot body, UUID storyId) {
-    UUID loggedUserId = getLoggerUserId();
     
-    // TODO: Permission check
+    TokenizerType tokenizerType = body.getTokenizer();
+    String content = body.getContent();
+    String hint = body.getHint();
     
     fi.metatavu.metamind.persistence.models.Story story = storyController.findStoryById(storyId);
     if (story == null) {
       return createBadRequest(String.format("Story %s not found", storyId)); 
     }
     
+    fi.metatavu.metamind.persistence.models.Knot knot = storyController.createKnot(body.getType(), tokenizerType, body.getName(), content, hint, story, loggedUserId); 
+    
+    try {
+      UUID createdResourceId = authenticationController.createProtectedResource(loggedUserId, String.format(KNOT_NAME_BEGINNING, knot.getId()), String.format("/v2/stories/knots/%s", knot.getId()), "knot", SCOPES);
```

Comment:
Permissions should be defined only on story level, knot permissions are way too fine grained